### PR TITLE
Basic Layout #4

### PR DESCRIPTION
**Pull request by YarNik**

### CODE REVIEW

File: front/src/layouts/menu/menu.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+.el-menu.el-menu--horizontal {
+  border-bottom: 0px; 
+  height: 90px;   
+}
+
+.el-menu--horizontal>.el-menu-item {  
+  height: 90px;
+  padding: 15px 0px 0px;
+  margin: 0px 25px;
+  color: #FFFFFF;
```
Comment (horba) [inline front/src/layouts/menu/menu.css]: variable

File: front/src/layouts/menu/menu.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  font-size: 18px;   
+}
+
+.el-menu--horizontal>.el-menu-item.is-active {
+  border-bottom: 3px solid #FFFFFF;
+  color: #FFA21C;
```
Comment (horba) [inline front/src/layouts/menu/menu.css]: variable

File: front/src/styles/common.css [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.commonMainColorBackground {
+    background: #6663FF !important;
+}
+
+.commonLowerPartFooterBackground {
+    background: #5956FF !important;
```
Comment (horba) [inline front/src/styles/common.css]: no important at all

File: front/src/layouts/footer/footer.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <a>
+        <img alt="Facebook" src="../../assets/facebook.png">
+      </a>
+    </div>   
+  </div>
+  <div class="lowerPart commonLowerPartFooterBackground">
+    <span class="lowerPart allReles">
+      (с) Все права защищены
+    </span>
+    <el-button class="commonLowerPartFooterBackground">Пользовательское соглашение</el-button>
```
Comment (horba) [inline front/src/layouts/footer/footer.html]: class name in kebab case like common-part-footer-backround
this name seems strange, are you sure that it doesn't have some meaning, not structure name

File: front/src/layouts/footer/footer.css [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  font-size: 22px;
+  padding: 10px;
+}
+
+span.allReles {
+  margin-left: 150px;
+  font-size: 11px;
+}
+
+a:hover {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: it seems like common rule, not only for footer

File: front/src/layouts/footer/footer.css [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+div.socialNetworks {
+  margin-right: 115px; 
+}
+
+i {
+  font-size: 22px;
+  padding: 10px;
+}
+
+span.allReles {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: not specify div, span or other tag`s name if it hasn`t highly important difference

File: front/src/layouts/footer/footer.css [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+div.lowerPart {
+  height: 57px; 
+}
+
+button.privicyPolice {
+  margin-right: 140px;
+  font-size: 14px;
+}
+
+button {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: might it is more common case aand we have this type of button in other places?

File: front/src/layouts/footer/footer.css [hunk 8]
```diff
@@ -0,0 +1,74 @@
+div.body {
+  color: white;
+  width: 1440px;
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: variable

File: front/src/components/logo/logo.css [hunk 9]
```diff
@@ -0,0 +1,14 @@
+div.logo { 
+  font-style: normal;
+  font-weight: 600;
+  font-size: 21px;
+  line-height: 138%;   
+}
+
+.white {
+  color: #FFFFFF;
```
Comment (horba) [inline front/src/components/logo/logo.css]: variable?

**Review by horba — CHANGES_REQUESTED** [review]
kebab-case;
common styles;
variables;
delete unnecessary tag name specification;
no important!

File: front/src/layouts/footer/footer.css (hunk 8 above)
Comment (horba) [reply front/src/layouts/footer/footer.css]: 1) it should be variable (1440px), it is like breakpoints
2) it is not about body's width.

File: front/src/layouts/footer/footer.css [hunk 10]
```diff
@@ -0,0 +1,64 @@
+div.body {  
+  width: 1440px;
+}
+
+div.main, div.lowerPart {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: div.main,
div.lowerPart {

File: front/src/layouts/footer/footer.css [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+  margin-left: 150px;
+  font-size: 11px;
+}
+
+button.privicyPolice {
+  margin-right: 140px;
+  font-size: 14px;
+}
+
+button {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: is it only for footer or common rules?

File: front/src/layouts/footer/footer.css [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+div.logo {
+  margin: 5px 0px 10px 80px;
+  padding: 5px 0px 10px;   
+}
+
+div.stores {
+  margin-left: 80px;   
+}
+
+div.lowerPart {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: lower-part

File: front/src/layouts/footer/footer.css [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  font-size: 14px;
+}
+
+button { 
+  border: 0px;
+  padding: 10px;
+  font-size: 16px;
+}
+
+div.socialNetworks {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: social-networks

File: front/src/layouts/footer/footer.html [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+      </el-link>
+      <el-link>
+        <img alt="Twitter" src="../../assets/twitter.png">
+      </el-link>
+      <el-link>
+        <img alt="Facebook" src="../../assets/facebook.png">
+      </el-link>
+    </div>   
+  </div>
+  <div class="lowerPart footer-color-background">
```
Comment (horba) [inline front/src/layouts/footer/footer.html]: lowerPart? it is unclear for me. might footer-bottom?

File: front/src/layouts/footer/footer.css [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+div.stores {
+  margin-left: 80px;   
+}
+
+div.lowerPart {
+  height: 57px; 
+}
+
+div.lowerPart > span {
```
Comment (horba) [inline front/src/layouts/footer/footer.css]: it is too specific and unflexible, i am not sure that you need class lower-part at all

File: front/src/layouts/footer/footer.html [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    <span>
+      (с) Все права защищены
+    </span>
+    <el-button class="footer-color-background main-color footer">Пользовательское соглашение</el-button>
+    <el-button class="footer-color-background main-color privicyPolice">Privicy Police</el-button>
```
Comment (horba) [inline front/src/layouts/footer/footer.html]: why buttons? if it is link to some pages it should be link. it doesn't have any common with buttons

File: front/src/layouts/menu/menu.css [hunk 17]
```diff
@@ -0,0 +1,28 @@
+div.main {
+  width: 1440px;
```
Comment (horba) [inline front/src/layouts/menu/menu.css]: again?)
menu couldn't be static width. look on fb, github, are they static width?

File: front/src/styles/variables.css [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+    background-color: #5956FF;
+}
+
+ul.main-color > li.main-color {
+    color: white;
+}
+
+ul.main-color > li.main-color.is-active {
+    border-bottom: 3px solid white;
+    color: #FFA21C;
```
Comment (horba) [inline front/src/styles/variables.css]: em, it is css rules, might they could be ... in common.css
but variables are variables - https://www.w3schools.com/css/css3_variables.asp

File: front/src/views/unregisteredUser/unregisteredUser.html [hunk 19]
```diff
@@ -0,0 +1,5 @@
+<div>    
+  <Menu />
```
Comment (horba) [inline front/src/views/unregisteredUser/unregisteredUser.html]: <menu/>
<footer/>

kebab-case and no br

File: front/src/views/unregisteredUser/unregisteredUser.js [hunk 20]
```diff
@@ -0,0 +1,8 @@
+import { Menu, Footer } from '@layouts';
+
+export default {
+  name: 'UnregisteredUser',
+  components: {
+    Menu, Footer
+  }
+};
```
Comment (horba) [inline front/src/views/unregisteredUser/unregisteredUser.js]: view part should be in file name

File: front/src/views/moderator/moderator.vue [hunk 21]
```diff
@@ -0,0 +1,5 @@
+<template src="./moderator.html"></template>
+
+<script src="./moderator.js"></script>
+
+<style scoped src="./moderator.css"></style>
```
Comment (horba) [inline front/src/views/moderator/moderator.vue]: we don\t need moderator functionality yet

File: front/src/views/contentManager/contentManager.vue [hunk 22]
```diff
@@ -0,0 +1,5 @@
+<template src="./contentManager.html"></template>
+
+<script src="./contentManager.js"></script>
+
+<style scoped src="./contentManager.css"></style>
```
Comment (horba) [inline front/src/views/contentManager/contentManager.vue]: file name with view

**Review by horba — CHANGES_REQUESTED** [review]
kebab-style
naming
css variables
srtyles improves

File: front/src/router/index.js [hunk 23]
```diff
@@ -1,14 +1,39 @@
...
+import { Authorize, UnregisteredUser, RegisteredUser, Admin, Moderator, TextManager } from '@views';
 
 Vue.use(VueRouter);
 
 const routes = [
   {
     path: '/',
-    name: 'Home',
-    component: Home
+    name: 'UnregisteredUser',
```
Comment (horba) [inline front/src/router/index.js]: guest

File: front/src/router/index.js [hunk 24]
```diff
@@ -1,14 +1,39 @@
...
+    component: UnregisteredUser
+  },
+  {
+    path: '/authorize',
+    name: 'Authorize',
+    component: Authorize
+  },
```
Comment (horba) [inline front/src/router/index.js]: it can be deleted: it will be added in task of authorization

File: front/src/router/index.js [hunk 25]
```diff
@@ -1,14 +1,39 @@
...
+  {
+    path: '/registeredUser',
+    name: 'RegisteredUser',
```
Comment (horba) [inline front/src/router/index.js]: I don't sure that now we need separate view for unauthorized user. we can delete it now and return to this question after authorization implementation

File: front/src/router/index.js [hunk 26]
```diff
@@ -1,14 +1,39 @@
...
+    name: 'Admin',
+    component: Admin
+  },
+  {
+    path: '/moderator',
+    name: 'Moderator',
+    component: Moderator
+  },
+  {
+    path: '/contentManager',
```
Comment (horba) [inline front/src/router/index.js]: path: '/content-manager'

File: front/src/router/index.js [hunk 27]
```diff
@@ -1,14 +1,39 @@
...
+    name: 'ContentManager',
+    component: TextManager
   }
```
Comment (horba) [inline front/src/router/index.js]: it is not unnecessary now (admin, moderator, content manager)

File: front/src/router/index.js (hunk 23 above)
Comment (horba) [reply front/src/router/index.js]: And one more thing: is it functionality only for guests? Actually it is common for all users. So, it is common functionality for all users, so it is not correct to say that it is guest or unregistered (why un**registered**, what about registered but unauthorized users)

**Review by horba — CHANGES_REQUESTED** [review]
I am not sure that we need different views for guest and unauthorized user. Now it seems like overflow.  

I propose save but change Home component. and change it in simple way:
<menu-layout/>
<router-view></router-view>
<footer-layout/>

File: front/src/layouts/footer/footer.css (hunk 15 above)
Comment (YarNik) [reply front/src/layouts/footer/footer.css]: so, the bottom of the footer is a different color

File: front/src/views/view-user/view-user.view.js [hunk 28]
```diff
@@ -0,0 +1,8 @@
+import { Menu, Footer } from '@layouts';
+
+export default {
+  name: 'User',
+  components: {
+    'ff-menu': Menu, 'ff-footer': Footer
```
Comment (horba) [inline front/src/views/view-user/view-user.view.js]: you could specify correct name in menu and footer components via name property

File: front/src/layouts/layout-menu/layout-menu.layout.vue [hunk 29]
```diff
@@ -0,0 +1,7 @@
+<template src="./layout-menu.layout.html"></template>
+
+<script src="./layout-menu.layout.js"></script>
+
+<style scoped src="./layout-menu.layout.css"></style>
+
+<style src="../../styles/common.css"></style>
```
Comment (horba) [inline front/src/layouts/layout-menu/layout-menu.layout.vue]: I don\t understand why you add common css in each view, if we included it in root component (app.vue)

File: front/src/layouts/layout-menu/layout-menu.layout.js [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  data () {
+    return {
+      activeIndex: '1',
+      activeIndex2: '1'
+    };
+  },
+  methods: {
+    handleSelect (key, keyPath) {
+      console.log(key, keyPath);
```
Comment (horba) [inline front/src/layouts/layout-menu/layout-menu.layout.js]: non console in prod

File: front/src/views/view-user/view-user.view.js (hunk 28 above)
Comment (YarNik) [reply front/src/views/view-user/view-user.view.js]: but how to make "export {default as ff-footer}" in index.js?

File: front/src/App.vue [hunk 31]
```diff
@@ -7,6 +7,8 @@
   </div>
 </template>
 
+<style src="./styles/common.css"></style>
+
```
Comment (horba) [inline front/src/App.vue:12]: why didn't include variables.css?

File: front/src/components/component-logo/component-logo.component.css [hunk 32]
```diff
@@ -0,0 +1,10 @@
+div.logo { 
+  font-style: normal;
+  font-weight: 600;
+  font-size: 21px;
+  line-height: 138%;
```
Comment (horba) [inline front/src/components/component-logo/component-logo.component.css]: no percents as said before

File: front/src/components/component-logo/component-logo.component.vue [hunk 33]
```diff
@@ -0,0 +1,7 @@
+<template src="./component-logo.component.html"></template>
+
+<script src="./component-logo.component.js"></script>
+
+<style scoped src="./component-logo.component.css"></style>
+
+<style src="../../styles/common.css"></style>
```
Comment (horba) [inline front/src/components/component-logo/component-logo.component.vue]: why do you include it again?

File: front/src/layouts/layout-footer/layout-footer.layout.css [hunk 34]
```diff
@@ -0,0 +1,60 @@
...
+
+div.stores {
+  margin-left: 80px;   
+}
+
+div.footer-color-background {
+  height: 57px; 
+}
+
+span {
```
Comment (horba) [inline front/src/layouts/layout-footer/layout-footer.layout.css]: either it is scoped styles it is really bad way to manage margin and font via changing style for span

File: front/src/layouts/layout-footer/layout-footer.layout.js [hunk 35]
```diff
@@ -0,0 +1,8 @@
+import { Logo } from '@components';
+
+export default {
+  name: 'ff-footer',
+  components: {
+    'ff-logo': Logo
```
Comment (horba) [inline front/src/layouts/layout-footer/layout-footer.layout.js]: the same as with footer, change in logo component name into ff-logo and you don't need specify this name here in declarations of outer components

File: front/src/views/view-user/view-user.view.js (hunk 28 above)
Comment (horba) [reply front/src/views/view-user/view-user.view.js]: name property in component declaration and name of class gor this component are absolutely different things.

in index.js:
`export {default as Footer}"`

view-user.view.js:
```
import { Menu, Footer } from '@layouts';
export default {
  name: 'User',
  components: {
    Menu, Footer
}
```

user.view.html 
`<ff-footer/>`

footer.layout.js
```
export default {
  name: 'ff-footer'
}
```

File: front/src/views/view-user/view-user.view.vue [hunk 36]
```diff
@@ -0,0 +1,5 @@
+<template src="./view-user.view.html"></template>
+
+<script src="./view-user.view.js"></script>
+
+<style scoped src="./view-user.view.css"></style>
```
Comment (horba) [inline front/src/views/view-user/view-user.view.vue]: user.view.html/css/js
why did you add the view before user?

File: front/src/layouts/layout-footer/layout-footer.layout.vue [hunk 37]
```diff
@@ -0,0 +1,5 @@
+<template src="./layout-footer.layout.html"></template>
+
+<script src="./layout-footer.layout.js"></script>
+
+<style scoped src="./layout-footer.layout.css"></style>
```
Comment (horba) [inline front/src/layouts/layout-footer/layout-footer.layout.vue]: footer.layout.html/css/js

no layout before footer

File: front/src/styles/common.css [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+ul.color-background {
+    background-color: #6663FF;
+}
+
+button.second-color {
+    color: #6663FF;
+}
+
+.footer-color-background,
+button.footer-color-background {
```
Comment (horba) [inline front/src/styles/common.css]: did you read about css variables? Please, add the color values as variables into variables.css and use these variables instead of static text values in these classes

File: front/src/layouts/layout-menu/layout-menu.layout.vue [hunk 39]
```diff
@@ -0,0 +1,5 @@
+<template src="./layout-menu.layout.html"></template>
+
+<script src="./layout-menu.layout.js"></script>
+
+<style scoped src="./layout-menu.layout.css"></style>
```
Comment (horba) [inline front/src/layouts/layout-menu/layout-menu.layout.vue]: menu.layout.html/css/js

no layout before menu

**Review by horba — CHANGES_REQUESTED** [review]

**Review by horba — APPROVED** [review]
